Scripts/PAIF-N/01-deploy-vcf-workload-domain.ps1

```diff
@@ -0,0 +1,345 @@
+                     )
+                  }
+                  @{
+                     Name = "$DomainName-vds02"
+                     IsUsedByNsxt = $true
+                  }
+               )
+               NsxClusterSpec = @{
+                  NsxTClusterSpec = @{
+                     GeneveVlanId = 0
+                  }
+               }
+            }
+            Name = "$DomainName-cl01"
+         }
+      )
+   }
+   NsxTSpec = @{
+      NsxManagerSpecs = @(
+         @{
+            Name = "$domainName-nsx01a"
+            NetworkDetailsSpec = @{
+               DnsName = "$domainName-nsx01a.$domain"
+               IpAddress = "10.0.0.41"
+               SubnetMask = "255.255.255.0"
+               Gateway = $gateway
+            }
+         }
+         @{
+            Name = "$domainName-nsx01b"
+            NetworkDetailsSpec = @{
+               DnsName = "$domainName-nsx01b.$domain"
+               IpAddress = "10.0.0.42"
+               SubnetMask = "255.255.255.0"
+               Gateway = $gateway
+            }
+         }
+         @{
+            Name = "$domainName-nsx01c"
+            NetworkDetailsSpec = @{
+               DnsName = "$domainName-nsx01c.$domain"
+               IpAddress = "10.0.0.43"
+               SubnetMask = "255.255.255.0"
+               Gateway = $gateway
+            }
+         }
+      )
+      FormFactor = 'large'
+      LicenseKey = 'XXXXX-XXXXX-XXXXX-XXXXX-XXXXX'
+      NSXManagerAdminPassword = "VMware123!123"
+      Vip = '10.0.0.44'
+      VipFqdn = "$domainName-nsx01.$domain"
+   }
+}
+# --------------------------------------------------------------------------------------------------------------------------
+
+## Workload domain spec construction
+$esxiFqdns = $esxiHosts | Select-Object -ExpandProperty Fqdn
+$VcfHost = Invoke-VcfGetHosts -status "UNASSIGNED_USEABLE" | `
+   Select-Object -ExpandProperty Elements | `
+   Where-Object { $esxiFqdns -contains $_.Fqdn }
+
+$ComputeSpec = Initialize-VcfComputeSpec -ClusterSpecs (
+   $domainSpec.ComputeSpec.ClusterSpecs | ForEach-Object {
+      $_clusterSpec = $_
+
+      Initialize-VcfClusterSpec `
+         -DatastoreSpec (
+            Initialize-VcfDatastoreSpec `
+               -VsanDatastoreSpec (
+                  Initialize-VcfVsanDatastoreSpec `
+                     -DatastoreName $_.DatastoreSpec.VSanDatastoreSpec.DatastoreName `
+                     -FailuresToTolerate $_.DatastoreSpec.VSanDatastoreSpec.FailuresToTolerate `
+                     -LicenseKey $_.DatastoreSpec.VSanDatastoreSpec.LicenseKey
+               )) `
+         -HostSpecs (
+            $VcfHost | ForEach-Object {
+               $esxiHost = $esxiHosts | Where-Object { $_.Fqdn -eq $_.Fqdn } | Select-Object -First 1
+
+               Initialize-VcfHostSpec `
+                  -Id $_.Id `
+                  -LicenseKey $esxiHost.LicenseKey `
+                  -HostNetworkSpec (
+                     Initialize-VcfHostNetworkSpec -VmNics @(
```

Comment:
Does this require that the ESX hosts have 4 NICs? If so, document this expectation and if it can be relaxed

Reply:
2 of the NICs are used for Frontend/Management networks and the other 2 workload